code/clue.js

```diff
@@ -1,59 +1,374 @@
+    }, 2500)
+  }
+};
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+// This function will be invoked when you click the weapon card.
+const pickWeapon = () => {
+  if (killerIsPicked && !weaponIsPicked) {
+    document.getElementById('loaderWeapon').style.opacity = 1
+    setTimeout(function() {
+      document.getElementById('loaderWeapon').style.opacity = 0
+      let chanceFavWeapon = chanceOfFavouriteWeapon();
+      mystery.weapon = randomSelector(chanceFavWeapon)
+      document.getElementById('weaponName').innerHTML = `Type: ${mystery.weapon.name}`
+      document.getElementById('weaponWeight').innerHTML = `Weight: ${mystery.weapon.weight}`
+      weaponIsPicked = true
+    }, 2500)
+  }else alert('Found out who the killer is first');
+};
+
+//This function will be invoked when you click  the room card.
+const pickRoom = () => {
+  if (weaponIsPicked && !roomIsPicked) {
+    document.getElementById('loaderRoom').style.opacity = 1
+    setTimeout(function() { 
+      document.getElementById('loaderRoom').style.opacity = 0
+      let chanceForSuspectedRoom = chanceOfRoom();
+      mystery.room = randomSelector(chanceForSuspectedRoom) 
+      document.getElementById('roomName').innerHTML = `Where murder was commited: ${mystery.room.name}`
+      roomsIsPicked = true
+    },2500)   
+  } else {
+    alert('Look at the other cards first!')
+  }
+};
+
+//select a random weapon from the weapons array and assigns it to suspects favouriteWeapon. 
+const shuffleFavoriteWeapon = () => {
+  suspects.forEach(suspect => {
+    let randomWeapon = randomSelector(weapons).id;
+    suspect.favouriteWeapon = randomWeapon
+  });
+};
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+//function to increase the probabilty of favourite weapon to be picked. 
+const chanceOfFavouriteWeapon = () => {
+  const favWeapon = weapons.find(weapon => weapon.id === mystery.killer.favouriteWeapon);
+  const favWeaponArray = new Array(7).fill(favWeapon);
+  const calculatedWeapon = [...weapons, ...favWeaponArray];
+  return calculatedWeapon;
+};
+
+//function to increase the probability that the room is picked where weapon was found.
+const chanceOfRoom = () => {
+  const suspectedRoom = rooms.find(room => room.name === mystery.weapon.foundIn);
+  const suspectedRoomArray = new Array(7).fill(suspectedRoom);
+  const calculatedRooms = [...rooms, ...suspectedRoomArray];
+  return calculatedRooms;
+};
+
+//revealMystery  will be invoked when you click the 'reveal the crime' button.
+const revealMystery = () => {
+  if (
+    mystery.killer === null ||
+    mystery.weapon === null ||
+    mystery.room === null 
```

Comment:
Or we could write it shorter way :
```
if (!mystery.killer || !mystery.weapon || !mystery.room) { ... }
```

Reply:
Thank you, a lot cleaner! I have changed that now.